31 basic card modal

MorganBare opened the pull request.

MorganBare committed:
Added folder and files for the card modal to start

MorganBare committed:
Added basic structure for the card modal

MorganBare committed:
Merged main to branch

MorganBare committed:
Merge branch 'main' of https://github.com/chingu-voyages/v40-geckos-team-12 into 31-basic-card-modal

MorganBare committed:
Added Kebab icon and padding to modal container

MorganBare committed:
Added modal backdrop color and light gray/second font color to the themes. Updated values on cardModal to use the theme and not be hardcoded

MorganBare committed:
Added hardcoded checkboxes to the card modal. Added some stlying for dark/light mode

MorganBare committed:
Added basic structure and some styling for the status dropdown menu

MorganBare committed:
Knocked out most of the checkbox and modal with Lloyd

MorganBare committed:
Checkboxes now have their own state and are keeping track of each subtasks checked status and striking out the text for completed subtasks

MorganBare committed:
Checkboxes have been styled

MorganBare committed:
Dropdown & checkboxes are fully functional with a hardcoded task in the initial state

MorganBare committed:
Added toggle for card modal

LloydRivers reviewed (changes requested):
in the cardModal  folder you have a component called cardModal.jsx but react components must start with an uppercase first letter (double check the the others).

Under this you have cardModal.styled but on line 19 in components/cardModal.jsx on line 19 you are importing them from CardModal.styled so the warning in the console says the names don't match.

To my knowledge styled component files also have to start with an uppercase letter

MorganBare committed:
Rename cardModal.jsx to CardModal.jsx

MorganBare committed:
Rename cardModal.styled.js to CardModal.styled.js

MorganBare commented:
Names have been updated. I had both files starting with an uppercase letter in my VS code but somehow this was never updated here on GitHub.

LloydRivers committed:
removed duplicated theme variables.

LloydRivers committed:
create selectors

We need a way for each column to display the correct cards, since we have 3 states the cards can be in, I am looping through the entire array and filtering out the ones I want to show in each column.

LloydRivers committed:
add value to the options for filtering

LloydRivers committed:
remove map

Because I have changed some of the logic, the map is no longer needed.

LloydRivers committed:
use the selectors

I have imported the selectors I made earlier. These selectors will give me back 3 different array. Each array will then be used to show the data in the columns. I then send each array down as a prop (line 35, 39 and 44)

LloydRivers committed:
send the object upwards

So here, I have an onclick and I also know that since this component is being rendered from a loop, whatever card I click will be unique. Since it is unique I can pass it in a function and use the data in the modal. That way whatever modal is clicked will have the right data.

LloydRivers committed:
lots of red

So most of the red you see here is not what I have deleted, it is because I had a code formatted plugin.

Remember that dispatch sends an >>> action <<<, and an action is a function.

Here you are trying to send an action as well as another function called setCurrentTask.

LloydRivers committed:
change colors

When I clicked the card the BG of the modal was not right and the logic for the opacity was back to front.

LloydRivers committed:
add a map

This maps through all the cards
